bootstrap/cmd/bootstrap/app/router.go

```diff
@@ -40,6 +39,47 @@ type kfctlRouter struct {
 	namespace string
 }
 
+type BasicAuth struct {
+	Username string `json:"username,omitempty"`
+	Password string `json:"password,omitempty"`
+}
+
+type IAP struct {
+	OAuthClientId     string `json:"oAuthClientId,omitempty"`
+	OAuthClientSecret string `json:"oAuthClientSecret,omitempty"`
+}
+
+// For GCP cloud endpoint
+type EndpointConfig struct {
+	BasicAuth BasicAuth `json:"basicAuth,omitempty"`
+	IAP       IAP       `json:"iap,omitempty"`
+}
+
+// KfctlRequest is the request body for CreateDeployment and GetLatestStatus API
+type KfctlRequest struct {
```

Comment:
Lets call this something like ClickToDeployRequest
To indicate that its an internal request.  And lets expand the comment to explain that this is an "internal" API only for click to deploy front end.